### PR TITLE
Support different extensions for "config" and "addons" files

### PR DESCRIPTION
**Pull request by igor-dv**
Issue: #3737

## What I did
When one uses TS (not only, though), she will probably prefer using `.ts` in all of the files in a codebase. 
Currently, we are forcing people to use `config.js` and `addons.js` to configure their stuff.
I've used the same approach as in #3785 to fix this.

Notice, I've needed to wrap `.ts` transformer in jest to the following pattern:
```
ts-transformer -> babel-transformer
```
in order to support `require.context` in TS  with Jest 🤕

### CODE REVIEW

**Comment by igor-dv** [comment]
I have a strange issue with this test:

image

I've committed the snapshot but now it's failing in CircleCI only 🤷‍♂️ 

I am not sure why it's failed in the beginning though...

Anyone with Mac/Linux can try my branch + run the tests?

**Comment by ndelangen** [comment]
I'll check it out

File: lib/core/src/server/config.js [hunk 1]
```diff
@@ -2,8 +2,8 @@ import findCacheDir from 'find-cache-dir';
 import { logger } from '@storybook/node-logger';
 import { createDefaultWebpackConfig } from './config/defaults/webpack.config';
 import devBabelConfig from './config/babel';
-import loadBabelConfig from './babel_config';
-import loadCustomConfig from './loadCustomWebpackConfig';
+import loadBabelConfig from './loadCustomBabelConfig';
```
[the remark this answers is not in the document]
Comment (igor-dv) [reply lib/core/src/server/config.js]: Sure, will rename

**Comment by igor-dv** [comment]
its's green now... Dafuq merge from master did?

**Comment by ndelangen** [comment]
@igor-dv apply additional magic

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
LGTM

**Pull request merged**